types/types.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+
+	_ "github.com/go-sql-driver/mysql" // Load MySQL Driver
+)
+
+type JSONText []string
+
+func (j JSONText) Value() (driver.Value, error) {
+	b, err := json.Marshal(j)
+	if err != nil {
+		return nil, err
+	}
+	return string(b), nil
+}
+
+func (j *JSONText) Scan(value interface{}) error {
+	if value == nil {
+		return nil
+	}
+
+	var source []byte
+	switch t := value.(type) {
+	case []uint8:
+		source = t
+	default:
```

Comment:
Interesting. Is this just defence-in-depth or do we really need to pass a `[]uint8` slice to `json.Unmarshal`?

Reply:
Copied from the sqlx example, i think as we know the type we can probably just unmarsel it without checking the type, I am kind of new to go, so i was hoping a more experienced member can comment on this :)

Reply:
Yes, I agree. This looks much better to me:

```go
func (j *JSONText) Scan(value interface{}) error {
	if value == nil {
		return nil
	}

	buf, ok := value.([]byte)
	if !ok {
		return errors.New("received value is not a byte slice")
	}

	return json.Unmarshal(buf, j)
}
```